ui/imports/shared/views/chat/CompactMessageView.qml

```diff
@@ -259,41 +259,68 @@ Item {
         Loader {
             id: pinnedRectangleLoader
             active: !editModeOn && pinnedMessage
-            anchors.left: chatName.left
-            anchors.top: parent.top
-            anchors.topMargin: active ? Style.current.halfPadding : 0
 
-            sourceComponent: Component {
+            anchors.left: messageContent.left
+            anchors.top: chatImage.top
+            anchors.topMargin: active ? -2 : 0
+
+            sourceComponent: Item {
+                height: 24
+                width: childrenRect.width + Style.current.smallPadding
+
                 Rectangle {
-                    id: pinnedRectangle
-                    height: 24
-                    width: childrenRect.width + Style.current.smallPadding
-                    color: Style.current.pinnedRectangleBackground
+                    anchors.fill: parent
+
+                    color: Qt.rgba(Style.current.pinnedRectangleBackground.r,
+                                   Style.current.pinnedRectangleBackground.g,
+                                   Style.current.pinnedRectangleBackground.b, 1)
+                    opacity: Style.current.pinnedRectangleBackground.a
```

Comment:
Hmm, I fail to see the purpose of this :)

```
Qt.rgba(Style.current.pinnedRectangleBackground.r,
                                   Style.current.pinnedRectangleBackground.g,
                                   Style.current.pinnedRectangleBackground.b, 1)
``` 
gives you exactly `Style.current.pinnedRectangleBackground` color

Reply:
Not exactly. It gives me `Style.current.pinnedRectangleBackground` but with alpha set to `1`. The color is fully opaque. The trick is to use opaque color for both rectangles, and apply opacity on top using `opacity` property (with `layer.enabled: true`).